pkgs/build-support/node/fetch-pnpm-deps/pnpm-config-hook.sh

```diff
@@ -0,0 +1,44 @@
+# shellcheck shell=bash
+
+pnpmConfigHook() {
+    echo "Executing pnpmConfigHook"
+
+    if [ -n "${pnpmRoot-}" ]; then
+      pushd "$pnpmRoot"
+    fi
+
+    if [ -z "${pnpmDeps-}" ]; then
+      echo "Error: 'pnpmDeps' should be set when using pnpmConfigHook."
+      exit 1
+    fi
+
+    echo "Configuring pnpm store"
+
+    export HOME=$(mktemp -d)
+    export STORE_PATH=$(mktemp -d)
+
+    cp -Tr "$pnpmDeps" "$STORE_PATH"
+    chmod -R +w "$STORE_PATH"
+
+    pnpm config set store-dir "$STORE_PATH"
+
+    echo "Installing dependencies"
+
+    if ! pnpm install --offline --frozen-lockfile --ignore-script; then
+      echo
+      echo "ERROR: pnpm failed to install dependencies"
+      echo
+
+      exit 1
+    fi
```

Comment:
Shouldn't we fail anyway if the command fails and the error message doesn't seem to add much.